src/Microsoft.Performance.SDK/Processing/ProcessingSource.cs

```diff
@@ -254,6 +279,43 @@ public virtual void DisposeProcessor(ICustomDataProcessor processor)
             IEnumerable<IDataSource> dataSources,
             IProcessorEnvironment processorEnvironment,
             ProcessorOptions options);
+        
+        /// <summary>
+        ///     When implemented in a derived class, creates a new
+        ///     instance implementing <see cref="ICustomDataProcessor"/>
+        ///     to process the specified <paramref name="dataSourceGroup"/>.
+        /// </summary>
+        /// <param name="dataSourceGroup">
+        ///     The <see cref="IDataSourceGroup"/> to be processed by the processor.
+        ///     This parameter is guaranteed to be non-null, and all
+        ///     elements in the collection are guaranteed to be non-null.
+        /// </param>
+        /// <param name="processorEnvironment">
+        ///     The environment for this specific processor instance.
+        /// </param>
+        /// <param name="options">
+        ///     The options to pass to the processor.
+        /// </param>
+        /// <returns>
+        ///     A new <see cref="ICustomDataProcessor"/>. It is an error
+        ///     to return <c>null</c> from this method.
+        /// </returns>
+        protected virtual ICustomDataProcessor CreateProcessorCore(
+            IDataSourceGroup dataSourceGroup,
+            IProcessorEnvironment processorEnvironment,
+            ProcessorOptions options)
+        {
+            if (this is IDataSourceGrouper)
+            {
+                throw new InvalidOperationException(
+                    $"Prior to V2, you must override the {nameof(CreateProcessorCore)} which accepts a {nameof(IDataSourceGroup)} if you implement {nameof(IDataSourceGrouper)}");
```

Comment:
>if you implement [](http://example.com/codeflow?start=130&length=16)

nittiest nit of all nits (feel free to ignore): "if you implement" -> "when implementing"